iothub/device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/IotHubConnectionString.java

```diff
@@ -103,6 +116,16 @@ else if (attr.toLowerCase().startsWith(GATEWAY_HOST_NAME_ATTRIBUTE.toLowerCase()
             {
                 this.gatewayHostName = attr.substring(GATEWAY_HOST_NAME_ATTRIBUTE.length());
             }
+            else if (attr.toLowerCase().startsWith(MQTT_GATEWAY_HOST_NAME_ATTRIBUTE.toLowerCase()))
+            {
+                this.mqttGatewayHostName = attr.substring(MQTT_GATEWAY_HOST_NAME_ATTRIBUTE.length());
+            }
+        }
+
+        if (gatewayHostName != null && !gatewayHostName.isEmpty()
+                && mqttGatewayHostName != null && !mqttGatewayHostName.isEmpty())
+        {
+            throw new IllegalArgumentException("[GatewayHostName] and [MqttGatewayHostName] should NOT be specified at the same time.");
         }
```

Comment:
Throw an `IllegalArgumentException` when specifying [GatewayHostName] and [MqttGatewayHostName] at the same time.